maps-compose/src/main/java/com/google/maps/android/compose/Marker.kt

```diff
@@ -719,8 +719,8 @@ private fun AdvancedMarkerImpl(
                 }
             }
 
-            set(visible) { this.marker.isVisible = it }
-            set(zIndex) { this.marker.zIndex = it }
+            update(visible) { this.marker.isVisible = it }
+            update(zIndex) { this.marker.zIndex = it }
```

Comment:
I don't see these properties being set during initial AdvancedMarker construction, so this change looks invalid to me as is.

The original code here initializes the properties to default values, then "updates" them to something else via `set()`. I'd think that it would be more efficient to directly initialize them to the desired values at construction time instead. This is what is done everywhere else.

 At that point these `update()`s will do the right thing.